aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -237,10 +237,10 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
 
   // If dst is contiguous and there is no byte offset, we can save directly the result of
   // gather into dst. This reduces the overhead of doing an additional blit for most cases
-  bool returnGatherOutput = (dst_.is_contiguous() && !dst_byte_offset);
+  bool returnGatherOutput = (dst_.is_contiguous() && !dst_byte_offset && src_.dtype() == dst_.dtype());
   Tensor src;
 
-  if (!src_.is_contiguous()) {
+  if (src_.is_view() || !src_.is_contiguous()) {
```

Comment:
Does this have any perf. implications for the current benchmarks? I think previously for the views we'd solve them using the blit, but now since they could return early from gather, I wonder if this is improving some other networks

Reply:
I'll check that and will let you know.